### ISSUE TITLE
LuaRocks cannot install md5 in lua5.4 version.

### ISSUE THREAD

**Issue by bz-e** [luarocks/luarocks#1718]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - Windows/Linux/MacOS.
MacOS
- **LuaRocks version:**
  - `luarocks --version`
3.11.1
- **Configuration file:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find
    the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.
default configuration
- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.
luarocks install md5         
Warning: falling back to wget - install luasec >= 1.1 to get native HTTPS support
Installing https://luarocks.org/md5-1.3-1.rockspec

Error: Error fetching file: Failed downloading https://github.com/keplerproject/md5/archive/1.3.tar.gz - failed downloading https://github.com/keplerproject/md5/archive/1.3.tar.gz

- **Description:**
  - The steps you took in plain English to reproduce the problem.
When I use the command to install the md5 package, it shows an error that it cannot be downloaded. See the error above for details.

**Comment by FractalU** [comment]
I was able to install `md5` on Lua 5.4 without issues. I tested it with LuaRocks 3.9.2 on Windows Msys2 and with LuaRocks 3.11.1 on Haiku. At first glance it looks like the `wget` tool isn't able to download https://github.com/keplerproject/md5/archive/1.3.tar.gz. I got this impression from you output.
```
luarocks install md5
Warning: falling back to wget - install luasec >= 1.1 to get native HTTPS support
Installing https://luarocks.org/md5-1.3-1.rockspec
```
My guess is that the redirect didn't work properly. Then https://github.com/keplerproject/md5/archive/1.3.tar.gz redirects to https://github.com/lunarmodules/md5/archive/1.3.tar.gz.

Another possibility is that that `wget` or `curl` on you MacOS isn't able to make `https` connection with `https://github.com/`. It might be that `wget` or `curl` doesn't have newer certificates. A lot of `https` errors happen to not having newer certificates. Https has the certificate truest system to verify connections. With https, the server gives one or more certificates to indicated that it's truly that server and not someone else. If you don't have the newer certificates, it has a good chance that certain `https` connections won't be made because you don't trust these servers, simply you don't know their certificates. These certificates are foreign to you.

LuaRocks was able to get the rockspec from https://luarocks.org/md5-1.3-1.rockspec, but it failed to download https://github.com/keplerproject/md5/archive/1.3.tar.gz. That link was in the rockspec, so Luarocks was definitely able to get that rockspec. It could very well be that `wget` or `curl` on your MacOS was able to make a connection to `https://luarocks.org/` but not `https://github.com/` due to certificate issues explained above.

It would be great if there is more feedback. Can you test the following commands on your terminal and report back? It's best to make a folder via `mkdir <folder name>` and go to it via `cd <folder name>`. After that you can delete that folder.
```
wget https://github.com/
```
```
wget https://github.com/keplerproject/md5/archive/1.3.tar.gz
```
```
curl -o index.html https://github.com/
```
```
curl -L -O https://github.com/keplerproject/md5/archive/1.3.tar.gz
```
**Btw**. you can edit your original report to make it much more readable. There is a lot of information that is not meant to be in the bug report such `Please ...`, `The steps you took ...`, etc. Those are merely placeholders.
